controllers/vspheredeploymentzone_controller.go

```diff
@@ -144,11 +133,23 @@ func (r vsphereDeploymentZoneReconciler) Reconcile(ctx context.Context, request
 			logr.Error(err, "patch failed", "vsphereDeploymentZone", vsphereDeploymentZoneContext.String())
 		}
 	}()
-
 	if !vsphereDeploymentZone.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, r.reconcileDelete(vsphereDeploymentZoneContext)
 	}
 
+	// Fetch the VSphereFailureDomain after checking if this VSphereDeploymentZone is being deleted. Otherwise the
```

Comment:
Hm but this means VSphereFailureDomain will never be set for reconcileDelete which means we now have dead code in reconcileDelete?

Reply:
Wouldn't it be better to handle the not found case differently depending on regular / delete case? (i.e. ignore the error on delete)

Reply:
Good idea!